airbyte-integrations/bases/base-python/base_python/cdk/utils/event_timing.py

```diff
@@ -0,0 +1,97 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+import datetime
+import time
+from contextlib import contextmanager
+from dataclasses import dataclass, field
+from threading import Lock, local
+
+from base_python.logger import AirbyteLogger
+
+logger = AirbyteLogger()
+
+
+class EventTimer:
+    """Simple nanosecond resolution event timer for debugging, initially intended to be used to record streams execution
+    time for a source.
+       As it is likely that streams could be read in parallel, we use a thread local storage to store the
+       stack of events and thus keep track of the current event.
+       Event nesting follows a LIFO pattern, so finish will apply to the last started event.
+    """
+
+    thread_local = local()
```

Comment:
I agree with @avida here, thread safety wasn't a requirement.

Reply:
Removed this